client/hijack.go

```diff
@@ -64,11 +60,6 @@ func fallbackDial(proto, addr string, tlsConfig *tls.Config) (net.Conn, error) {
 }
 
 func (cli *Client) setupHijackConn(ctx context.Context, req *http.Request, proto string) (net.Conn, string, error) {
-	req.URL.Host = cli.addr
-	if cli.proto == "unix" || cli.proto == "npipe" {
-		// Override host header for non-tcp connections.
-		req.Host = DummyHost
-	}
```

Comment:
Was this supposed to be included in this PR? Seems to unrelated to `Client.buildRequest`.

Reply:
This is the logic that was duplicated, and I was trying to centralise; https://github.com/moby/moby/blob/ee794231240b4f4087cffb434dcc67b06f402a47/client/request.go#L99-L117

Reply:
Ah, right! :facepalm: